python/cuml/dask/ensemble/randomforestclassifier.py

```diff
@@ -370,21 +440,101 @@ def fit(self, X, y):
 
         return self
 
-    def predict(self, X):
+    def predict(self, X, deep_check=False):
         """
         Predicts the labels for X.
 
+        Parameters
+        ----------
+        X : dask cuDF dataframe or numpy array (n_samples, n_features)
+        deep_check : boolean (default = False)
```

Comment:
This might need a little bit more explanation or a more clear one. Is this for debugging , why is it only for binary, what does it mean to be extensive?